lib/install/writable.js

```diff
@@ -0,0 +1,11 @@
+'use strict'
+var fs = require('fs')
+// The Windows implementation of `fs.access` has a bug where it will
+// sometimes return access errors all the time for directories, even
+// when access is available. As all we actually test ARE directories, this
+// is a bit of a problem.
+// FIXME: When this is corrected in Node/iojs, update this check to be
+// a bit more specific
+var isWindows = process.platform === 'win32'
+
+module.exports = (fs.access && !isWindows) ? require('./writable-fs-access.js') : require('./writable-fs-open.js')
```

Comment:
I think it would probably be confusing to defactor this all into a single file for the purposes of npm. When would you use `writable-fs-access.js` and `writable-fs-open.js` independently?